addons/spectate/Displays.hpp

```diff
@@ -0,0 +1,362 @@
+                    idc = CAM_BG_IDC;
+
+                    x = QUOTE(0.6 * XFACTOR);
+                    y = QUOTE(0.4 * YFACTOR);
+                    w = QUOTE(7.5 * XFACTOR);
+                    h = QUOTE(2 * YFACTOR);
+
+                    colorBackground[] = {0,0,0,0.75};
+                };
+                class Free: RscButton {
+                    idc = CAM_FREE_IDC;
+
+                    onButtonClick = QUOTE([FREE_CAMERA] call FUNC(ui_changeCamera));
+
+                    x = QUOTE(1.3 * XFACTOR);
+                    y = QUOTE(0.8 * YFACTOR);
+                    w = QUOTE(1.63 * XFACTOR);
+                    h = QUOTE(1.37 * YFACTOR);
+
+                    text = CAM_ICON_FREE_SELECTED;
+                    tooltip = "$STR_A3_Spectator_free_camera_tooltip";
+
+                    style = 48;
+                    colorBackground[] = {0,0,0,0};
+                    colorBackgroundDisabled[] = {0,0,0,0};
+                    colorBackgroundActive[] = {0,0,0,0};
+                    colorFocused[] = {0,0,0,0};
+                };
+                class Third: RscButton {
+                    idc = CAM_THIRD_IDC;
+
+                    onButtonClick = QUOTE([THIRD_PERSON] call FUNC(ui_changeCamera));
+
+                    x = QUOTE(3.6 * XFACTOR);
+                    y = QUOTE(0.8 * YFACTOR);
+                    w = QUOTE(1.63 * XFACTOR);
+                    h = QUOTE(1.37 * YFACTOR);
+
+                    text = CAM_ICON_THIRD;
+                    tooltip = "$STR_A3_Spectator_3pp_camera_tooltip";
+
+                    style = 48;
+                    colorBackground[] = {0,0,0,0};
+                    colorBackgroundDisabled[] = {0,0,0,0};
+                    colorBackgroundActive[] = {0,0,0,0};
+                    colorFocused[] = {0,0,0,0};
+                };
+                class First: RscButton {
+                    idc = CAM_FIRST_IDC;
+
+                    onButtonClick = QUOTE([FIRST_PERSON] call FUNC(ui_changeCamera));
+
+                    x = QUOTE(5.8 * XFACTOR);
+                    y = QUOTE(0.8 * YFACTOR);
+                    w = QUOTE(1.63 * XFACTOR);
+                    h = QUOTE(1.37 * YFACTOR);
+
+                    text = CAM_ICON_FIRST;
+                    tooltip = "$STR_A3_Spectator_1pp_camera_tooltip";
+
+                    style = 48;
+                    colorBackground[] = {0,0,0,0};
+                    colorBackgroundDisabled[] = {0,0,0,0};
+                    colorBackgroundActive[] = {0,0,0,0};
+                    colorFocused[] = {0,0,0,0};
+                };
+            };
+        };
+        class MapGroup: RscControlsGroupNoScrollbars {
+            idc = MAP_GROUP_IDC;
+
+            x = QUOTE(-0.8 * XFACTOR + XOFFSET);
+            y = QUOTE(-0.8 * YFACTOR + YOFFSET);
+            w = QUOTE(41.5962 * XFACTOR);
+            h = QUOTE(26.7 * YFACTOR);
+
+            class controls {
+                class Map: RscMapControl {
```

Comment:
That's how it's handled in BI's EG spectate, can you do a video of the behavior and post your resolution?